examples/vision/xray_classification_with_tpus.py

```diff
@@ -0,0 +1,489 @@
+            tf.keras.layers.MaxPool2D(),
+            conv_block(32),
+            conv_block(64),
+            conv_block(128),
+            tf.keras.layers.Dropout(0.2),
+            conv_block(256),
+            tf.keras.layers.Dropout(0.2),
+            tf.keras.layers.Flatten(),
+            dense_block(512, 0.7),
+            dense_block(128, 0.5),
+            dense_block(64, 0.3),
+            tf.keras.layers.Dense(1, activation="sigmoid"),
+        ]
+    )
+
+    return model
+
+
+"""
+# 5. Correct for data imbalance
+
+We saw earlier in this notebook that the data was imbalanced, with more images classified
+as pneumonia than normal. We will correct for that in this following section.
+"""
+
+initial_bias = np.log([COUNT_PNEUMONIA / COUNT_NORMAL])
+initial_bias
+
+weight_for_0 = (1 / COUNT_NORMAL) * (TRAIN_IMG_COUNT) / 2.0
+weight_for_1 = (1 / COUNT_PNEUMONIA) * (TRAIN_IMG_COUNT) / 2.0
+
+class_weight = {0: weight_for_0, 1: weight_for_1}
+
+print("Weight for class 0: {:.2f}".format(weight_for_0))
+print("Weight for class 1: {:.2f}".format(weight_for_1))
+
+"""
+The weight for class `0` (Normal) is a lot higher than the weight for class `1`
+(Pneumonia). Because there are less normal images, each normal image will be weighted
+more to balance the data as the CNN works best when the training data is balanced.
+"""
+
+"""
+# 6. Train the model
+
+Since there are only two possible labels for the image, we will be using the
+`binary_crossentropy` loss. When we fit the model, identify the class weights. Because we
+are using a TPU, training will be relatively quick.
+
+For our metrics, we want to include precision and recall as they will provide use with a
+more informed picture of how good our model is. Accuracy tells us what fractions are the
+labels are correct. Since our data is not balanced, accuracy might give a skewed sense of
+a good model (i.e. a model that always predicts PNEUMONIA will be 74% accurate but is not
+a good model).
+
+Precision is the number of true positives (TP) over the sum of TP and false positives
+(FP). It shows what fraction of labeled positives are actually correct.
+
+Recall is the number of TP over the sum of TP and false negatves (FN). It shows what
+fraction of actual positives are correct.
+"""
+
+with strategy.scope():
+    model = build_model()
+
+    METRICS = [
+        "accuracy",
+        tf.keras.metrics.Precision(name="precision"),
+        tf.keras.metrics.Recall(name="recall"),
+    ]
+
+    model.compile(optimizer="adam", loss="binary_crossentropy", metrics=METRICS)
+
+"""
+From exploring the data and the model, I noticed that the training for the model has a
+slow start. However, after 25 epochs, the model slowly starts to converge.
+"""
+
+history = model.fit(
+    train_ds,
+    steps_per_epoch=TRAIN_IMG_COUNT // BATCH_SIZE,
+    epochs=EPOCHS,
+    validation_data=val_ds,
+    validation_steps=VAL_IMG_COUNT // BATCH_SIZE,
+    class_weight=class_weight,
+)
+
+"""
+# 7. Finetune the model
```

Comment:
Fine-tuning is normally done when you have a model trained on a different dataset, that you want to adapt to a new dataset. See for instance this guide https://keras.io/guides/transfer_learning/

Here you are training on the same dataset, with a different learning rate. You could simply add a learning rate schedule in the call to `fit()` above. Please remove this section, and instead  add a learning rate schedule to your original optimizer. See: https://keras.io/api/optimizers/learning_rate_schedules/